### PR TITLE
Feat/kg grpc client

### PR DESCRIPTION
**Pull request by angeltabar**

### CODE REVIEW

File: app/api/infrastructure/kgservice/service.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		}
+
+		var cat entity.KnowledgeGraphItemCat
+
+		switch value.Category {
+		case kgPaper:
+			cat = entity.KnowledgeGraphItemCatPaper
+		case kgRepo:
+			cat = entity.KnowledgeGraphItemCatCode
+		}
```
Comment (angelsanchez) [inline app/api/infrastructure/kgservice/service.go:67]: The KnowledgeGraphItemCat type is string.
The value.Category is string.
So you can use:
```
cat := entity.KnowledgeGraphItemCat(value.Category)
if !cat.IsValid() {
   return ....
}
```

File: app/api/infrastructure/kgservice/service.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	for i, value := range res.Items {
+		externalID := &value.ExternalId
+		if value.ExternalId == "" {
+			externalID = nil
+		}
+
+		framework := &value.Framework
+		if value.Framework == "" {
+			framework = nil
+		}
```
Comment (angelsanchez) [inline app/api/infrastructure/kgservice/service.go:58]: why not a function that returns the str pointer or nil if the string is empty:
```
func stringToPointer(s string) *string {
  if s == "" {
    return nil
  }
  return &s
}
```
with this helper func:
```
items[i] = entity.KnowledgeGraphItem{
....
			ExternalID:  stringToPointer(value.ExternalId),
			Framework:   stringToPointer(value.Framework),
		}
```

File: app/api/usecase/kg/interactor.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import (
+	"context"
+
+	"github.com/konstellation-io/kdl-server/app/api/entity"
+	"github.com/konstellation-io/kdl-server/app/api/infrastructure/kgservice"
+	"github.com/konstellation-io/kdl-server/app/api/pkg/logging"
+)
+
+// Interactor implements the UseCase interface.
+type Interactor struct {
```
Comment (angelsanchez) [inline app/api/usecase/kg/interactor.go:12]: Interactor should be private `interactor`

File: app/api/usecase/kg/interactor.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	logger    logging.Logger
+	kgService kgservice.KGService
+}
+
+// NewInteractor is a constructor function.
+func NewInteractor(logger logging.Logger,
+	kgService kgservice.KGService) *Interactor {
```
Comment (angelsanchez) [inline app/api/usecase/kg/interactor.go:19]: should return a interface `kg.UseCase`

**Review by angelsanchez — CHANGES_REQUESTED** [review]

**Review by marcvivancos — APPROVED** [review]

File: app/api/infrastructure/kgservice/service.go (hunk 1 above)
Comment (angeltabar) [reply app/api/infrastructure/kgservice/service.go]: value.Category comes in lowercase. The output is capitalized to validate it. We are not super sure if this is the best way to do it

**Review by angelsanchez — APPROVED** [review]

**Pull request merged**